### PR TITLE
Enhance Docker tooling

### PR DESCRIPTION
**Pull request by jorgecuesta**
## Note:

Review this PR after #17 

## Summary

Update docker configurations to improve environment handling

- Adjust `.dockerignore` to include specific `.env` files instead of generic `.env*`.
- Modify `docker-compose.[production|development].yml` to use optional environment variables for extra parameters.
- Update `.env.sample` with new environment variables and the latest `PGADMIN_VERSION`.
- Refactor `docker/pgadmin/servers.json` to simplify server naming and grouping.

## Issue

- Missing a way to pass extra parameters to SubQuery Node and Query process.
- PGadmin was not loading `servers.json` values.

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [x] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -14,7 +14,10 @@ src/types
 project.yaml
```
Comment (Olshansk) [inline .dockerignore:14]: In the future, when we have dependent PRs, please do the following

1. In the title, say: `[DO NOT MERGE UNTIL UPDATING BASE]...`
2. Make the base PR the other branch, not `main`

RIght now, I believe it reflects all the changes in the other PR as well.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: .dockerignore (hunk 1 above)
Comment (jorgecuesta) [reply .dockerignore:14]: 1. Ok
2. If I set other branch as base, so far I known, I can't change it later.

Comment (Olshansk) [reply .dockerignore:14]: Re 2:

https://github.com/user-attachments/assets/aedcb99c-c479-4f43-81ba-4a6c31cc2a49

Comment (jorgecuesta) [reply .dockerignore:14]: Updated

**Review by Olshansk — APPROVED** [review]

**Pull request merged**